pkgs/build-support/rust/build-rust-package/default.nix

```diff
@@ -52,8 +52,6 @@
 , buildAndTestSubdir ? null
 , ... } @ args:
 
-assert cargoVendorDir == null && cargoLock == null -> !(args ? cargoSha256) && !(args ? cargoHash)
-  -> throw "cargoSha256, cargoHash, cargoVendorDir, or cargoLock must be set";
```

Comment:
> This PR fixes a few problems ...
>
> Previously cargoHash/cargoSha256 would be required arguments, if both were unset buildRustPackage would throw an error and the user would have to set either to an empty string/lib.fakeHash. 

I don't agree that this is a problem. I think the throw is correct and is useful.

> This allows you to write an expression that "prefetches" the hashes with less boilerplate, which is always nice.

I don't see that there is any actual benefit from this.